### ISSUE TITLE
Input elements can't reliably be focused with hints

### ISSUE THREAD

**Issue by toofar** [qutebrowser/qutebrowser#7623]
**Version info**:
qutebrowser v2.5.2
Git commit: 0f0102757093-dirty on qt6-v2 (2023-03-15 17:07:31 +1300)
Backend: QtWebEngine 6.4.2, based on Chromium 102.0.5005.177 (from api)
Qt: 6.4.2
Also on the latest 6.5 beta

**Does the bug happen if you start with `--temp-basedir`?**:
Yep

**Description**
In some situations you can hint an input element, end up in insert mode, but the text you type doesn't appear in the text box.
The only way I've found so far is to undo a close tab, there might be other ways to get into this situation though?

**How to reproduce**
Save this HTML test page somehere:

```html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <title>Textarea</title>
    </head>
    <body>
        <p>These text input elements can't be hinted right away after undoing a
        tab close. Both `f` and `F` have different behaviours.</p>

        <p>`f` will land you in insert mode but the text box doesn't look
        selected and has no blinking cursor.</p>

        With `F` the text box looks focused and has a blinking cursor but the
          text you type still doesn't show up there.
        <br/>
        <textarea id="qute-textarea"></textarea>
        <br/>
        <textarea id="qute-textarea-2"></textarea>
        <br/>
        <form>
          <input type="text"></input>
        </form>
        <br/>
        <p>This button is still hintable in that state though. (Use <code>:config-dict-add --replace content.javascript.log info info</code> to see JS messages in the qutebrowser log output.)</p>
        <button onclick="console.log('clicky')">button</button>
    </body>
</html>
```

Host that somewhere like `python3 -m http.server 3366`.
Open qutebrowser with two tabs: `QUTE_QT_WRAPPER=PyQt6 python3 -m qutebrowser -T --debug about:blank http://localhost:3366/textarea.html`
Close the tab.
Undo that.
Use `f` or `F` to try to select one of the three text boxes.
See that you are in input mode.
Type some stuff.

Compare this to Qt5.15 where text you type reliably goes into the text box.

**Comment by toofar** [comment]
Looking at the diff of the debug logs from undoing a tab close, Qt5 on the left and Qt6 on the right:
![image](https://user-images.githubusercontent.com/7419144/225824293-dfff99e4-43bb-4b0e-bb75-3daaa49f71ad.png)

It looks like the additional actions on Qt6 are:
1. we get a tab event (oh no) (lines 24-31)
2. we get an extra child object
3. the focus shifts to `qutebrowser.mainwindow.mainwindow.MainWindow` (line 43)
4. the object create on line 20 is deleted

**Comment by toofar** [comment]
Turns out the issue was a symptom of a more complex situation. When a WebEnginePage losed its child widget it calls parent.focusNextPrevChild(true) to let its parent focus a new element. We ended up getting a MainWindow object focussed which I don't think should ever be getting focus in our setup. Which implies to me that we have some gaps in out focus proxies and layouts and such.

But addressing that isn't technically required for this issue, the WebEnginePage is just swapping out widgets, ideally that would be atomic (or some kind of seamless) instead of causing this whole mess. Also what do we focus when the current tab doesn't take focus?
So I'm just re-focussing the new tab which resolves this specific scenario. If we run into more issues like this hopefully the above investigation helps with something more comprehensive.
